Subject: [PATCH] feat(home): complete homepage sections and UI details
---
Index: app/Http/Controllers/Account/OrderController.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/Http/Controllers/Account/OrderController.php b/app/Http/Controllers/Account/OrderController.php
--- a/app/Http/Controllers/Account/OrderController.php	(revision 910a7ca410525486628986e5cf02398d3e3f7ece)
+++ b/app/Http/Controllers/Account/OrderController.php	(revision c06400b790ba09f952ba714ff075802a26893f4f)
@@ -15,11 +15,12 @@
 
         $withoutSlider = true;
         $withoutFooter = true;
+        $title = 'سفارشات';
         $userOrders = Order::query()
             ->where('user_id','=',Auth::id())
             ->with('orderItems.product')
             ->orderByDesc('created_at')
             ->paginate();
-        return view('account.orders',compact('withoutSlider','withoutFooter','userOrders'));
+        return view('account.orders',compact('withoutSlider','withoutFooter','userOrders','title'));
     }
 }
Index: app/Http/Controllers/Account/ProfileController.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/Http/Controllers/Account/ProfileController.php b/app/Http/Controllers/Account/ProfileController.php
--- a/app/Http/Controllers/Account/ProfileController.php	(revision 910a7ca410525486628986e5cf02398d3e3f7ece)
+++ b/app/Http/Controllers/Account/ProfileController.php	(revision c06400b790ba09f952ba714ff075802a26893f4f)
@@ -14,8 +14,9 @@
     {
         $withoutSlider = true;
         $withoutFooter = true;
+        $title = 'پروفایل';
         $user = auth()->user();
-        return view('account.profile', compact('withoutSlider', 'withoutFooter', 'user'));
+        return view('account.profile', compact('withoutSlider', 'withoutFooter', 'user','title'));
     }
 
     public function edit(ProfileEditRequest $request)
Index: app/Http/Controllers/CartController.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/Http/Controllers/CartController.php b/app/Http/Controllers/CartController.php
--- a/app/Http/Controllers/CartController.php	(revision 910a7ca410525486628986e5cf02398d3e3f7ece)
+++ b/app/Http/Controllers/CartController.php	(revision c06400b790ba09f952ba714ff075802a26893f4f)
@@ -11,8 +11,9 @@
     public function index()
     {
         $withoutSlider = true;
+        $title = 'سبد خرید';
         $userCart = CartService::getItemWithDetails();
-        return view('cart.index',compact('withoutSlider','userCart'));
+        return view('cart.index',compact('withoutSlider','userCart','title'));
     }
 
     public function add(Request $request)
Index: app/Http/Controllers/CheckoutController.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/Http/Controllers/CheckoutController.php b/app/Http/Controllers/CheckoutController.php
--- a/app/Http/Controllers/CheckoutController.php	(revision 910a7ca410525486628986e5cf02398d3e3f7ece)
+++ b/app/Http/Controllers/CheckoutController.php	(revision c06400b790ba09f952ba714ff075802a26893f4f)
@@ -13,8 +13,9 @@
     public function index()
     {
         $withoutSlider = true;
+        $title = 'ثبت سفارش';
         $userCart = CartService::getItemWithDetails();
-        return view('checkout.index',compact('withoutSlider','userCart'));
+        return view('checkout.index',compact('withoutSlider','userCart','title'));
     }
 
     public function store(CheckoutStoreRequest $request)
Index: app/Http/Controllers/IndexController.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/Http/Controllers/IndexController.php b/app/Http/Controllers/IndexController.php
--- a/app/Http/Controllers/IndexController.php	(revision 910a7ca410525486628986e5cf02398d3e3f7ece)
+++ b/app/Http/Controllers/IndexController.php	(revision c06400b790ba09f952ba714ff075802a26893f4f)
@@ -2,12 +2,25 @@
 
 namespace App\Http\Controllers;
 
+use App\Enums\ProductStatus;
+use App\Models\Product;
+use App\Models\ProductCategory;
 use Illuminate\Http\Request;
 
 class IndexController
 {
     public function index()
     {
-        return view('index');
+        $title = 'صفحه اصلی';
+        $products = Product::query()
+            ->where('status','=',ProductStatus::PUBLISHED)
+            ->orderByDesc('created_at')
+            ->paginate();
+        $productCategories = ProductCategory::get();
+        $bestSellingProduct = Product::withSum('orderItems','qty')
+            ->orderByDesc('order_items_sum_qty')
+            ->limit(5)
+            ->get();
+        return view('index',compact('title','products','productCategories','bestSellingProduct'));
     }
 }
Index: app/Http/Controllers/ProductController.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/Http/Controllers/ProductController.php b/app/Http/Controllers/ProductController.php
--- a/app/Http/Controllers/ProductController.php	(revision 910a7ca410525486628986e5cf02398d3e3f7ece)
+++ b/app/Http/Controllers/ProductController.php	(revision c06400b790ba09f952ba714ff075802a26893f4f)
@@ -12,6 +12,7 @@
     public function index()
     {
         $withoutSlider = true;
+        $title = 'فروشگاه';
         $products = Product::query()
             ->applyFilter()
             ->applySort()
@@ -19,7 +20,7 @@
             ->paginate()
             ->withQueryString();
         $productCategories = ProductCategory::get();
-        return view('product.index',compact('withoutSlider','products','productCategories'));
+        return view('product.index',compact('withoutSlider','products','productCategories','title'));
     }
 
     public function removeFilter()
@@ -35,9 +36,6 @@
         $product->load('productCategory');
         $productCategory = Product::all();
         $productCategory->load('productCategory');
-//        dd(
-//            $productCategory->toArray()
-//        );
         $withoutSlider = true;
         return view('product.layouts.show',compact('withoutSlider','product','productCategory'));
     }
Index: config/project.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/config/project.php b/config/project.php
new file mode 100644
--- /dev/null	(revision c06400b790ba09f952ba714ff075802a26893f4f)
+++ b/config/project.php	(revision c06400b790ba09f952ba714ff075802a26893f4f)
@@ -0,0 +1,5 @@
+<?php
+
+return [
+    'title'=>'فروشگاه دورنیکا'
+];
Index: resources/views/index.blade.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/resources/views/index.blade.php b/resources/views/index.blade.php
--- a/resources/views/index.blade.php	(revision 910a7ca410525486628986e5cf02398d3e3f7ece)
+++ b/resources/views/index.blade.php	(revision c06400b790ba09f952ba714ff075802a26893f4f)
@@ -27,22 +27,16 @@
         <div
             class="flex items-center justify-evenly flex-wrap mt-12 child:mb-8 gap-x-8 child:items-center child:flex-col child:duration-300 child:cursor-pointer child:gap-y-1 child:text-gray-800 child:dark:text-gray-300 child:relative"
         >
-            <a href="http://127.0.0.1:8000/products?category_id%5B1%5D=on" class="group flex">
-                <img src="./assets/images/category/5.png"
-                     class="w-[100px] h-[100px] lg:w-[120px] lg:h-[120px] object-cover group-hover:grayscale group-hover:opacity-90 duration-300"
-                     alt="category1" />
-                <p class="pt-1 text-sm lg:text-lg line-clamp-1">
-                    الکترونیک
-                </p>
-            </a>
-            <a href="http://127.0.0.1:8000/products?category_id%5B2%5D=on" class="group flex">
-                <img src="./assets/images/category/5.png"
-                     class="w-[100px] h-[100px] lg:w-[120px] lg:h-[120px] object-cover group-hover:grayscale group-hover:opacity-90 duration-300"
-                     alt="category1" />
-                <p class="pt-1 text-sm lg:text-lg line-clamp-1">
-                    کتاب
-                </p>
-            </a>
+            @foreach($productCategories as $category)
+                <a href="{{route('product.index',['category_id['.$category->id.']' => $category->id])}}" class="group flex">
+                    <img src="./assets/images/category/5.png"
+                         class="w-[100px] h-[100px] lg:w-[120px] lg:h-[120px] object-cover group-hover:grayscale group-hover:opacity-90 duration-300"
+                         alt="category1" />
+                    <p class="pt-1 text-sm lg:text-lg line-clamp-1">
+                        {{$category->name}}
+                    </p>
+                </a>
+            @endforeach
         </div>
     </section>
 
@@ -77,7 +71,7 @@
                         </svg>
                     </button>
                 </div>
-                <a href="http://127.0.0.1:8000/products?sort=newest"
+                <a href="{{route('product.index',generateSortRouteParameter('newest'))}}"
                    class="group shadow-xl text-sm md:text-base flex gap-x-1.5 items-center px-2 h-10 md:px-3 text-white bg-blue-600 rounded-xl">
                     <p>مشاهده همه</p>
                     <span
@@ -93,122 +87,9 @@
         <!-- Latest products Slider -->
         <div class="swiper LatestProducts mt-5 w-full">
             <div class="swiper-wrapper py-5">
-                <!-- PRODUCT ITEM -->
-                <div class="swiper-slide product-card group">
-                    <!-- product header -->
-                    <div class="product-card_header">
-                        <div class="flex items-center gap-x-2">
-                            <form action="http://127.0.0.1:8000/cart/add" method="POST">
-                                <input type="hidden" name="_token" value="4bYZgaUXw6NRUpehNCo79sVxnGyoZdTpQMBuSG48" autocomplete="off">
-                                <input type="hidden" name="product_id" value="1"/>
-                                <input type="hidden" name="qty" value="1"/>
-
-                                <div class="tooltip">
-                                    <button
-                                        type="submit"
-                                        class="rounded-full p-1.5 app-border app-hover"
-                                    >
-                                        <svg class="size-4">
-                                            <use href="#shopping-cart"></use>
-                                        </svg>
-                                    </button>
-                                    <div class="tooltiptext">
-                                        سبد خرید
-                                    </div>
-                                </div>
-                            </form>
-                        </div>
-                        <!-- badge offer -->
-                        <span class="product-card_badge">
-                10
-                %
-                تخفیف‌
-            </span>
-                    </div>
-                    <!-- product img -->
-                    <a href="http://127.0.0.1:8000/products/1">
-                        <img
-                            class="product-card_img group-hover:opacity-0 absolute"
-                            src="http://127.0.0.1:8000/assets/images/products/1.png"
-                            alt=""
-                        >
-                        <img class="product-card_img opacity-0 group-hover:opacity-100"
-                             src="http://127.0.0.1:8000/assets/images/products/1.png" alt="">
-                    </a>
-                    <!--  product footer -->
-                    <div class="space-y-2">
-                        <a href="http://127.0.0.1:8000/products/1" class="product-card_link">
-                            گوشی هوشمند | Smartphone
-                        </a>
-                        <!-- Rate and Price -->
-                        <div class="product-card_price-wrapper">
-                            <!-- Price -->
-                            <div class="product-card_price">
-                                <del>500,000 <h6>تومان</h6></del>
-                                <p>450,000</p>
-                                <span>تومان</span>
-                            </div>
-                        </div>
-                    </div>
-                </div>
-                <!-- PRODUCT ITEM -->
-                <div class="swiper-slide product-card group">
-                    <!-- product header -->
-                    <div class="product-card_header">
-                        <div class="flex items-center gap-x-2">
-                            <form action="http://127.0.0.1:8000/cart/add" method="POST">
-                                <input type="hidden" name="_token" value="4bYZgaUXw6NRUpehNCo79sVxnGyoZdTpQMBuSG48" autocomplete="off">
-                                <input type="hidden" name="product_id" value="2"/>
-                                <input type="hidden" name="qty" value="1"/>
-
-                                <div class="tooltip">
-                                    <button
-                                        type="submit"
-                                        class="rounded-full p-1.5 app-border app-hover"
-                                    >
-                                        <svg class="size-4">
-                                            <use href="#shopping-cart"></use>
-                                        </svg>
-                                    </button>
-                                    <div class="tooltiptext">
-                                        سبد خرید
-                                    </div>
-                                </div>
-                            </form>
-                        </div>
-                        <!-- badge offer -->
-                        <span class="product-card_badge">
-                1
-                %
-                تخفیف‌
-            </span>
-                    </div>
-                    <!-- product img -->
-                    <a href="http://127.0.0.1:8000/products/2">
-                        <img
-                            class="product-card_img group-hover:opacity-0 absolute"
-                            src="http://127.0.0.1:8000/assets/images/products/1.png"
-                            alt=""
-                        >
-                        <img class="product-card_img opacity-0 group-hover:opacity-100"
-                             src="http://127.0.0.1:8000/assets/images/products/1.png" alt="">
-                    </a>
-                    <!--  product footer -->
-                    <div class="space-y-2">
-                        <a href="http://127.0.0.1:8000/products/2" class="product-card_link">
-                            رمان | Novel
-                        </a>
-                        <!-- Rate and Price -->
-                        <div class="product-card_price-wrapper">
-                            <!-- Price -->
-                            <div class="product-card_price">
-                                <del>80,000 <h6>تومان</h6></del>
-                                <p>79,000</p>
-                                <span>تومان</span>
-                            </div>
-                        </div>
-                    </div>
-                </div>
+               @foreach($products as $product)
+                   @include('component.product')
+               @endforeach
             </div>
         </div>
     </section>
@@ -244,7 +125,7 @@
                         </svg>
                     </button>
                 </div>
-                <a href="http://127.0.0.1:8000/products?sort=best_selling"
+                <a href="{{route('product.index',generateSortRouteParameter('best_selling'))}}"
                    class="group shadow-xl text-sm md:text-base flex gap-x-1.5 items-center px-2 h-10 md:px-3 text-white bg-blue-600 rounded-xl">
                     <p>مشاهده همه</p>
                     <span
@@ -261,121 +142,9 @@
         <div class="swiper BestSelling mt-5 w-full">
             <div class="swiper-wrapper py-5">
                 <!-- PRODUCT ITEM -->
-                <div class="swiper-slide product-card group">
-                    <!-- product header -->
-                    <div class="product-card_header">
-                        <div class="flex items-center gap-x-2">
-                            <form action="http://127.0.0.1:8000/cart/add" method="POST">
-                                <input type="hidden" name="_token" value="4bYZgaUXw6NRUpehNCo79sVxnGyoZdTpQMBuSG48" autocomplete="off">
-                                <input type="hidden" name="product_id" value="1"/>
-                                <input type="hidden" name="qty" value="1"/>
-
-                                <div class="tooltip">
-                                    <button
-                                        type="submit"
-                                        class="rounded-full p-1.5 app-border app-hover"
-                                    >
-                                        <svg class="size-4">
-                                            <use href="#shopping-cart"></use>
-                                        </svg>
-                                    </button>
-                                    <div class="tooltiptext">
-                                        سبد خرید
-                                    </div>
-                                </div>
-                            </form>
-                        </div>
-                        <!-- badge offer -->
-                        <span class="product-card_badge">
-                10
-                %
-                تخفیف‌
-            </span>
-                    </div>
-                    <!-- product img -->
-                    <a href="http://127.0.0.1:8000/products/1">
-                        <img
-                            class="product-card_img group-hover:opacity-0 absolute"
-                            src="http://127.0.0.1:8000/assets/images/products/1.png"
-                            alt=""
-                        >
-                        <img class="product-card_img opacity-0 group-hover:opacity-100"
-                             src="http://127.0.0.1:8000/assets/images/products/1.png" alt="">
-                    </a>
-                    <!--  product footer -->
-                    <div class="space-y-2">
-                        <a href="http://127.0.0.1:8000/products/1" class="product-card_link">
-                            گوشی هوشمند | Smartphone
-                        </a>
-                        <!-- Rate and Price -->
-                        <div class="product-card_price-wrapper">
-                            <!-- Price -->
-                            <div class="product-card_price">
-                                <del>500,000 <h6>تومان</h6></del>
-                                <p>450,000</p>
-                                <span>تومان</span>
-                            </div>
-                        </div>
-                    </div>
-                </div>
-                <!-- PRODUCT ITEM -->
-                <div class="swiper-slide product-card group">
-                    <!-- product header -->
-                    <div class="product-card_header">
-                        <div class="flex items-center gap-x-2">
-                            <form action="http://127.0.0.1:8000/cart/add" method="POST">
-                                <input type="hidden" name="_token" value="4bYZgaUXw6NRUpehNCo79sVxnGyoZdTpQMBuSG48" autocomplete="off">
-                                <input type="hidden" name="product_id" value="2"/>
-                                <input type="hidden" name="qty" value="1"/>
-
-                                <div class="tooltip">
-                                    <button
-                                        type="submit"
-                                        class="rounded-full p-1.5 app-border app-hover"
-                                    >
-                                        <svg class="size-4">
-                                            <use href="#shopping-cart"></use>
-                                        </svg>
-                                    </button>
-                                    <div class="tooltiptext">
-                                        سبد خرید
-                                    </div>
-                                </div>
-                            </form>
-                        </div>
-                        <!-- badge offer -->
-                        <span class="product-card_badge">
-                1
-                %
-                تخفیف‌
-            </span>
-                    </div>
-                    <!-- product img -->
-                    <a href="http://127.0.0.1:8000/products/2">
-                        <img
-                            class="product-card_img group-hover:opacity-0 absolute"
-                            src="http://127.0.0.1:8000/assets/images/products/1.png"
-                            alt=""
-                        >
-                        <img class="product-card_img opacity-0 group-hover:opacity-100"
-                             src="{{asset('assets/images/products/1.png')}}" alt="">
-                    </a>
-                    <!--  product footer -->
-                    <div class="space-y-2">
-{{--                        <a href="http://127.0.0.1:8000/products/2" class="product-card_link">--}}
-                            رمان | Novel
-                        </a>
-                        <!-- Rate and Price -->
-                        <div class="product-card_price-wrapper">
-                            <!-- Price -->
-                            <div class="product-card_price">
-                                <del>80,000 <h6>تومان</h6></del>
-                                <p>79,000</p>
-                                <span>تومان</span>
-                            </div>
-                        </div>
-                    </div>
-                </div>
+                @foreach($bestSellingProduct as $product)
+                    @include('component.product')
+                @endforeach
             </div>
         </div>
     </section>
Index: resources/views/layouts/app.blade.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/resources/views/layouts/app.blade.php b/resources/views/layouts/app.blade.php
--- a/resources/views/layouts/app.blade.php	(revision 910a7ca410525486628986e5cf02398d3e3f7ece)
+++ b/resources/views/layouts/app.blade.php	(revision c06400b790ba09f952ba714ff075802a26893f4f)
@@ -5,8 +5,11 @@
     <meta charset="UTF-8"/>
     <meta name="viewport" content="width=device-width, initial-scale=1.0"/>
     <title>
-        فروشگاه درنیکا
-        | صفحه اصلی
+        {{config('project.title')}}
+        @isset($title)
+        |
+            {{$title}}
+        @endisset
     </title>
 
     <link rel="stylesheet" href="{{asset('assets/styles/app.css')}}">
Index: resources/views/layouts/header.blade.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/resources/views/layouts/header.blade.php b/resources/views/layouts/header.blade.php
--- a/resources/views/layouts/header.blade.php	(revision 910a7ca410525486628986e5cf02398d3e3f7ece)
+++ b/resources/views/layouts/header.blade.php	(revision c06400b790ba09f952ba714ff075802a26893f4f)
@@ -100,14 +100,16 @@
                     <svg class="size-6">
                         <use href="#shopping-bag"/>
                     </svg>
+                    @if(\App\Services\CartService::cartItemCount() > 0)
                     <span class="absolute -top-1 -right-1 flex h-4 w-4">
                         <span class="animate-ping absolute inline-flex h-full w-full rounded-full bg-red-600 opacity-75">
 
                         </span>
-                        <span class="relative inline-flex rounded-full h-4 w-4 bg-red-500 text-xs pt-1 flex-center text-white">
-                            1
+                            <span class="relative inline-flex rounded-full h-4 w-4 bg-red-500 text-xs pt-1 flex-center text-white">
+                            {{\App\Services\CartService::cartItemCount()}}
                         </span>
                     </span>
+                    @endif
                 </a>
             </div>
         </div>
